src/pac.c

```diff
@@ -112,6 +112,26 @@ pac_may_have_muzzy(pac_t *pac) {
 	return pac_decay_ms_get(pac, extent_state_muzzy) != 0;
 }
 
+#ifdef LIMIT_USIZE_GAP
+size_t pac_batched_alloc_retained_size(size_t size) {
+	if (size > SC_LARGE_MAXCLASS) {
+		return size;
+	}
+	size_t x = lg_floor((size<<1)-1);
+	size_t lg_delta = (x < SC_LG_NGROUP + LG_QUANTUM + 1)
+	    ?  LG_QUANTUM : x - SC_LG_NGROUP - 1;
+	size_t delta = ZU(1) << lg_delta;
+	size_t delta_mask = delta - 1;
+	size_t caching_size = (size + delta_mask) & ~delta_mask;
```

Comment:
This and above are redundant logic from the size class compute, right?

Reply:
Yes, made a separate function for this part to prevent redundancy.